python/cugraph/components/test_cc.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+from itertools import product
+import queue
+import time
+
+import numpy as np
+import pytest
+from scipy.io import mmread
+
+import cudf
+import cugraph
+from librmm_cffi import librmm as rmm
+from librmm_cffi import librmm_config as rmm_cfg
+
+# Temporarily suppress warnings till networkX fixes deprecation warnings
+# (Using or importing the ABCs from 'collections' instead of from
+# 'collections.abc' is deprecated, and in 3.8 it will stop working) for
+# python 3.7.  Also, this import networkx needs to be relocated in the
+# third-party group once this gets fixed.
+import warnings
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import networkx as nx
+
+
+print('Networkx version : {} '.format(nx.__version__))
+
+
+def read_mtx_file(mm_file):
+    print('Reading ' + str(mm_file) + '...')
+    return mmread(mm_file).asfptype()
+
+
+def read_csv_file(mm_file):
+    print('Reading ' + str(mm_file) + '...')
+    return cudf.read_csv(mm_file, delimiter=' ',
+                         dtype=['int32', 'int32', 'float32'], header=None)
+
+def networkx_call(M):
+    M = M.tocsr()
+    if M is None:
+        raise TypeError('Could not read the input graph')
+    if M.shape[0] != M.shape[1]:
+        raise TypeError('Shape is not square')
+
+    Gnx = nx.DiGraph(M)
+
+    # Weakly Connected components call:
+    print('Solving... ')
+    t1 = time.time()
+
+    # same parameters as in NVGRAPH
+    result = nx.weakly_connected_components(Gnx)
+    t2 = time.time() - t1
+
+    print('Time : ' + str(t2))
+
+    labels = sorted(result)
+    return labels
+
+
+def cugraph_call(cu_M):
+    # Device data
+    sources = cu_M['0']
+    destinations = cu_M['1']
+
+    # cugraph Pagerank Call
+    G = cugraph.Graph()
+    G.add_edge_list(sources, destinations, None)
+    t1 = time.time()
+    df = cugraph.weak_cc(G)
+    t2 = time.time() - t1
+    print('Time : '+str(t2))
+
+    result = df['labels'].to_array()
+
+    labels = sorted(result)
+    return labels
+
+# these should come w/ cugraph/python:
+#
+DATASETS = ['../datasets/dolphins'] #,
+#            '../datasets/coPapersDBLP',     # missing
```

Comment:
we should use the same data sets as in other python tests (eg. netscience etc.).
Dolphin runs in only one SM.

Reply:
Please specify exactly what you want run here. I cannot guess what you'd like. Pagerank only runs `dolphins` and `karate`. If `karate` is symmetric, I can add it.

Reply:
Please replace

```
# these should come w/ cugraph/python:
#
DATASETS = ['../datasets/dolphins'] #,
#            '../datasets/coPapersDBLP',     # missing
#            '../datasets/coPapersCiteseer', # missing
#            '../datasets/hollywood']        # missing
```
by

```
DATASETS = ['../datasets/dolphins',
                '../datasets/netscience']
```

Reply:
Okay, added `karate` to the test.

Reply:
I think there's a typo in your latest update. As discussed above it should be :

DATASETS = ['../datasets/dolphins',
                '../datasets/netscience']

Reply:
The test fails with netscience. I'm in the process of fixing that.

Reply:
Okay, pushed fix for the bug.